sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/Models/TimeSeriesId.cs

```diff
@@ -0,0 +1,157 @@
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 1 string key.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1)
+        {
+            _key1 = (true, stringKey1);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 2 string keys.
+        /// </summary>
+        /// <param name="stringKey1">The first key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp2">The second key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 2 string keys.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1, string timeSeriesIdStringProp2)
+        {
+            _key1 = (true, stringKey1);
+            _key2 = (true, timeSeriesIdStringProp2);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 3 string keys.
+        /// </summary>
+        /// <param name="timeSeriesIdStringProp1">The first key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp2">The second key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp3">The third key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 3 string keys.
+        /// </remarks>
+        public TimeSeriesId(string timeSeriesIdStringProp1, string timeSeriesIdStringProp2, string timeSeriesIdStringProp3)
+        {
+            _key1 = (true, timeSeriesIdStringProp1);
+            _key2 = (true, timeSeriesIdStringProp2);
+            _key3 = (true, timeSeriesIdStringProp3);
+        }
+
+        /// <summary>
+        /// Builds an array to represent a single Time Series Id in order to use it when making calls against the Time
+        /// Series Insights client library.
+        /// </summary>
+        /// <returns>An array representing a single Time Series Id.</returns>
+        public object[] ToArray()
+        {
+            if (_key3.Item1)
+            {
+                return new object[] { _key1.Item2, _key2.Item2, _key3.Item2 };
+            }
+
+            if (_key2.Item1)
+            {
+                return new object[] { _key1.Item2, _key2.Item2 };
+            }
+
+            return new object[] { _key1.Item2 };
+        }
+
+        /// <summary>
+        /// Builds a string representation of the Time Series Id.
+        /// </summary>
+        /// <returns>The keys that make up the Time Series Id separated by a comma.</returns>
+        public string GetId()
+        {
+            if (_key3.Item1)
+            {
+                return $"{_key1.Item2},{_key2.Item2},{_key3.Item2}";
```

Comment:
Are they supposed to be comma delimited? I wondered. :)

Reply:
The reason why I chose this format, because that's how they are represented in the Time Series Insights portal. So wanted to remain consistent when representing the Id.

Reply:
Sounds like we should check with the TSI team to be sure.

Reply:
Checked with the TSI team. Even though this is how it is represented in the UX, it's possible that a user might have a 'comma' in the Id. Changed to avoid confusion in case the user has such an Id in their TSI